sdk/core/Azure.Core/src/ResourceIdentifier.cs

```diff
@@ -526,7 +527,7 @@ public static ResourceIdentifier Parse(string input)
         /// If the method returns false, result will be null.
         /// </param>
         /// <returns> True if the parse operation was successful; otherwise, false. </returns>
-        public static bool TryParse(string input, out ResourceIdentifier? result)
+        public static bool TryParse(string input, [MaybeNullWhen(false)] out ResourceIdentifier? result)
```

Comment:
Tested this to validate and apparently the warning always exists if the out var is nullable so the attribute seems to make no difference.

If we change to

```c#
public static bool TryParse(string input, [MaybeNullWhen(false)] out ResourceIdentifier result)
```

Then it seems to work as expected.  I don't now the backwards compatibility rules on nullable annotation changes.  @KrzysztofCwalina is there another link I should be using besides https://learn.microsoft.com/en-us/dotnet/core/compatibility/library-change-rules to check?

My guess is going from `string` to `string?` is fine but going from `string?` to `string` is not?

```c#
namespace nullable
{
    internal class Program
    {
        static void Main(string[] args)
        {
            if (Foo.TryParse("x", out var y))
                Console.WriteLine(y.Length); //does not give warning

            Foo.TryParse("x", out var y2);
            Console.WriteLine(y2.Length); //gives warning cs8602
        }
    }

    internal class Foo
    {
        public static bool TryParse(string x, [MaybeNullWhen(false)] out string y)
        {
            y = null;
            return false;
        }
    }
}
```